bundler/lib/bundler/dsl.rb

```diff
@@ -308,6 +309,20 @@ def add_git_sources
         repo_name ||= user_name
         "https://#{user_name}@bitbucket.org/#{user_name}/#{repo_name}.git"
       end
+      
+      git_source(:gitlab) do |repo_name|
```

Comment:
Since "gitlab" is name of the project and it could be also self-hosted and this points to public instance of gitlab.com, should this be `gitlab_com`?

Reply:
> Isn't that also true about GitHub enterprise?

GitHub EE is not so common per my experience, but gitlab is often self-hosted by organizations. I'm also not happy about github shortcut btw.

Reply:
> I'm also not happy about github shortcut btw. 

I agree. We were moving in the way of declaring it within the Gemfile.

This PR is a small ask overall. We've now drawn a clear line and we're enforcing it. I think this will be the last addition for a while baring a major shift.

We should think about, and even purposely deprecate the others that are smaller. That will give us a pattern for deprecating these when they are too small. I'd accept one no deprecating bitbucket, provided it printed the exact code to add to your Gemfile to fix the deprecation.

Reply:
Sorry, I misremembered the bitbucket traffic numbers. Scratch that. Anything smaller should be removed though.

Reply:
> I think this will be the last addition for a while barring a major shift.

:shrug: